src/utils.cpp

```diff
@@ -849,10 +857,17 @@ resolve_binary_path(const std::string &cmd, const char *env_paths, int pid)
     if (pid > 0 && pid_in_different_mountns(pid))
       rel_path = path_for_pid_mountns(pid, path);
     else
-      rel_path = path;
+      rel_path = realpath(path);
+
     if (bcc_elf_is_exe(rel_path.c_str()) ||
         bcc_elf_is_shared_obj(rel_path.c_str()))
-      valid_executable_paths.push_back(rel_path);
+    {
+      auto it = std::find(valid_executable_paths.begin(),
+                          valid_executable_paths.end(), rel_path);
+      if (it == valid_executable_paths.end()) {
+        valid_executable_paths.push_back(rel_path);
```

Comment:
Instead of doing this manually, make `valid_executable_paths` a `std::unordered_set`.